File: archive/java/muse/algorithms/sorting/SelectionSort.java
package muse.algorithms.sorting;

public final class SelectionSort {

  private SelectionSort() {}

  public static <T extends Comparable<T>> void sort(T[] arr) {
    if (arr.length == 0) {
      return;
    }

    T temp;

    int iMin;

    for (int i = 0, bound = arr.length - 1; i < bound; i++) {
      iMin = i;
      for (int j = i + 1; j <= bound; j++) {
        if (arr[j].compareTo(arr[iMin]) < 0) {
          iMin = j;
        }
      }
      if (iMin != i) {
        temp = arr[i];
        arr[i] = arr[iMin];
        arr[iMin] = temp;
      }
    }
  }
}

File: archive/php/muse/algorithms/sorting/SelectionSort.php
<?php

declare(strict_types=1);

namespace muse\algorithms\sorting\SelectionSort;

function sort(array &$arr): void
{
    if (count($arr) == 0) {
        return;
    }

    $iMin = 0;

    for ($i = 0, $bound = count($arr) - 1; $i < $bound; $i++) {
        $iMin = $i;
        for ($j = $i + 1; $j <= $bound; $j++) {
            if ($arr[$j] < $arr[$iMin]) {
                $iMin = $j;
            }
        }
        if ($iMin != $i) {
            [$arr[$i], $arr[$iMin]] = [$arr[$iMin], $arr[$i]];
        }
    }
}
